### PR TITLE
Update to Android Studio 3.5

### PR DESCRIPTION
**Pull request by vinaysshenoy**
This PR bumps the Gradle plugin to 3.5.0. In addition, it updates libraries which use KAPT (Dagger, Room) to the latest versions and enabled KAPT incremental annotation processing.

### CODE REVIEW

File: spellfix/.gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /build
 .externalNativeBuild
+.cxx
```
Comment (sanchita) [inline spellfix/.gitignore]: What is this for?

Comment (sanchita) [inline spellfix/.gitignore]: Whats this for?

Comment (vinaysshenoy) [reply spellfix/.gitignore]: The new gradle plugin generates a new kind of native build artifact that we should ignore.

Comment (vinaysshenoy) [reply spellfix/.gitignore]: Just like `.externalNativeBuild`, it is autogenerated when the project is imported and should not be checked into VCS.

**Pull request merged**